### ISSUE TITLE
v1.4.0 : Bug with token authentication, don't upgrade from 1.3 !

### ISSUE THREAD

**Issue by chandon** [mailjet/mailjet-apiv3-php#144]
When trying to authenticate with token, i got this exception : 
    
    Cannot use a scalar value as an array
    vendor/mailjet/mailjet-apiv3-php/src/Mailjet/Client.php:335
    vendor/mailjet/mailjet-apiv3-php/src/Mailjet/Client.php:140
    vendor/mailjet/mailjet-apiv3-php/src/Mailjet/Client.php:61

I think that 
1. function _setTokenAuthetication should be renamed _setTokenAuthe**n**tication
2. Client.php:61 uses function _setTokenAuthetication with wrong parameters...

**Comment by chandon** [comment]
see #142 too

**Comment by jlambe** [comment]
Hello,
until now I was not able to use the MailJet PHP library in order to connect and send emails with the MailJet Send API.

Using the library, the client cannot authenticate. Also, when specifying an API version as `v3.1` on the client setting, on the debug message, we can see that the client is not using it as it sends requests to version `3` of the Send API instead.

The library is using the Guzzle package behind the scene. So here you can find a code snippet for those having issues with the MailJet library:
```php
$client = new \GuzzleHttp\Client([
    'base_uri' => 'https://api.mailjet.com/v3.1/'
]);

$messages = [
    [
        'From' => [
            'Email' => 'email@domain.com'
            'Name' => 'Company'
        ],
        'To' => [
            [
                'Email' => $to,
                'Name' => $name
            ]
        ],
        'Subject' => $subject,
        'TextPart' => 'Hello, here is an email',
        'HTMLPart' => '<h1>Hello</h1><p>Here is an email</p>'
    ]
];

$response = $client->post('send', [
    'auth' => [
        getenv('MJ_KEY'),
        getenv('MJ_SECRET')
    ],
    'json' => [
        'Messages' => $messages
    ]
]);
```
Hope this helps while issues are being fixed on the library.

**Comment by mskochev** [comment]
Hello all, sorry for this issue. We have released an update (v1.4.1) that should be solving the case.

The readme will be updated shortly with some use cases explaining how to work with basic auth and token auth.
